homeassistant/components/deconz/config_flow.py

```diff
@@ -182,27 +178,18 @@ def _update_entry(self, entry, host, port, api_key=None):
         if discovery_info[ssdp.ATTR_UPNP_MANUFACTURER_URL] != DECONZ_MANUFACTURERURL:
             return self.async_abort(reason="not_deconz_bridge")
 
-        uuid = discovery_info[ssdp.ATTR_UPNP_UDN].replace("uuid:", "")
-
-        _LOGGER.debug("deCONZ gateway discovered (%s)", uuid)
+        self.bridgeid = normalize_bridge_id(discovery_info[ssdp.ATTR_UPNP_SERIAL])
+        await self.async_set_unique_id(self.bridgeid)
 
         parsed_url = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION])
 
         for entry in self.hass.config_entries.async_entries(DOMAIN):
-            if uuid == entry.data.get(CONF_UUID):
-                if entry.source == "hassio":
-                    return self.async_abort(reason="already_configured")
+            if self.bridgeid == entry.unique_id and entry.source != "hassio":
```

Comment:
You still want to abort if the source == hassio.

Reply:
It gets aborted by the abort if unique id configured method afterwards